src/t8_cmesh/t8_cmesh_vertex_conn_tree_to_vertex.cxx

```diff
@@ -0,0 +1,166 @@
+                                     ? t8_cmesh_get_tree_class (cmesh_from, itree)
+                                     : t8_cmesh_get_ghost_class (cmesh_from, itree - num_local_trees);
+
+    const int num_tree_vertices = t8_eclass_num_vertices[tree_class];
+
+    /* Size of list must match number of tree vertices. */
+    SC_CHECK_ABORTF (
+      tree_indices.size () == (size_t) num_tree_vertices,
+      "ERROR. Number of mapped local tree vertices for tree %i does not equal number of tree vertices. %zu != %i\n",
+      itree, tree_indices.size (), num_tree_vertices);
+    /* We now build the array of global indices of this tree, sorted 
+     * by the trees local indices. */
+    t8_gloidx_t *global_tree_indices = T8_ALLOC (t8_gloidx_t, num_tree_vertices);
+    for (int ivertex = 0; ivertex < num_tree_vertices; ++ivertex) {
+      global_tree_indices[ivertex] = tree_indices[ivertex].first;
+    }
+    const t8_gloidx_t global_tree_id = t8_cmesh_get_global_id (cmesh_from, itree);
+    set_global_vertex_ids_of_tree_vertices (cmesh, global_tree_id, global_tree_indices, num_tree_vertices);
+    T8_FREE (global_tree_indices);
+  }
+}
+
+/** Set all global vertex ids of a local tree.
+  * \param [in] cmesh The considered cmesh
+  * \param [in] local_tree A local tree id of \a cmesh
+  * \param [in] global_vertex_id The ids of the global vertices in order of \a local_tree's vertices.
+  * \param [in] num_vertices Must match the number of vertices of \a local_tree
+  *
+  * \note Cmesh must not be committed.
+  */
+void
+t8_cmesh_vertex_conn_tree_to_vertex::set_global_vertex_ids_of_tree_vertices (const t8_cmesh_t cmesh,
+                                                                             const t8_gloidx_t global_tree,
+                                                                             const t8_gloidx_t *global_tree_vertices,
+                                                                             const int num_vertices)
+{
+  T8_ASSERT (t8_cmesh_is_initialized (cmesh));
+  T8_ASSERT (num_vertices >= 0);
+  T8_ASSERT (global_tree_vertices != NULL);
+
+  /* TODO: we currently do not check whether the num_vertices argument
+   *       matches the number of vertices of the tree.
+   *       We cannot do it here, since this function call happens before commit,
+   *       thus we might not even know the eclass of the tree.
+   *       Maybe it is possible to check this during t8_cmesh_add_attributes?
+   */
+
+  /* We copy the data directly, hence set data_persiss to 0 */
+  const int data_persists = 0;
+  t8_debugf ("Setting %i global vertices for global tree %li.\n", num_vertices, global_tree);
+  t8_cmesh_set_attribute_gloidx_array (cmesh, global_tree, t8_get_package_id (), T8_CMESH_GLOBAL_VERTICES_ATTRIBUTE_KEY,
+                                       global_tree_vertices, num_vertices, data_persists);
+  state = FILLED;
+}
+
+/* TODO: What if the attribute is not set? error handling */
+const t8_gloidx_t *
```

Comment:
It is already unusable from C since it is in a class, so we can also get rid of the C array.
```suggestion
const std::array<t8_gloidx_t>
```

Reply:
Tried this, does not work.

std::array is a constant (compile time) size array.
Since the number of vertices is determined at runtime, we cannot use it.

Since, we store the vertices as the raw data and return a pointer to this raw data to directly modify them, i argue to leave this be.

Alternatively: Return a vector that wraps around the storage. This will increase memory usage by adding the vector functionality and add runtime for constructing the wrapper.
But increases usability?
It will definitely increase type-safety.

Would try if i still have time at the end.